src/components/fields/image-upload/image-upload-helper.ts

```diff
@@ -0,0 +1,20 @@
+import { IImage } from "./types";
+
+export namespace ImageUploadHelper {
+	/**
+	 * picks an available slot
+	 * slot refers to the index of the file selected
+	 * this is for tracking so we can remove the correct file when user removes it
+	 * if maxFiles is not provided, assign based on no. of files / previous highest slot
+	 */
+	export const findAvailableSlot = (maxFiles: number, files: IImage[]) => {
```

Comment:
can check if slot is supposed to be unique and incrementing e.g. 0,1,2,3?

currently getting sequences like [undefined, 0], [0, 1, 0], [0, 0, 1, 1]

Reply:
that's a bug indeed, I have updated the function
- have the return value default to the derived max slot, it will be zero initially when nothing is set
- removed `maxFiles` argument, I want to avoid throwing errors so I removed it and the check for max images will be outside this function
- added unit tests for it